src/orient/rational-calc.orient

```diff
@@ -0,0 +1,354 @@
+    describe(encoding_window_time_parallel_asic, "Time to encode a window in parallel with ASICs", secs)
+    describe(decoding_time, "Time to decode a sector", secs)
+    describe(kdf_time, "Time to compute a KDF", secs)
+
+    encoding_time = encoding_window_time * windows
+    encoding_time_parallel = encoding_time / rig_cores
+
+    encoding_window_time = (window_nodes * graph_parents) * kdf_time * drg_layers
+    encoding_window_time_parallel = encoding_window_time / rig_cores
+
+    decoding_time = encoding_window_time
+
+    Asic:
+      encoding_time_parallel_asic = encoding_time_parallel / hashing_amax
+      encoding_time_asic = encoding_time / hashing_amax
+      encoding_window_time_asic = encoding_window_time / hashing_amax
+      encoding_window_time_parallel_asic = encoding_window_time_parallel / hashing_amax
+
+  Retrieval:
+    // window_read = rig_storage_read_mbs * window_size_mb
+
+  SNARK:
+    describe(porep_snark_constraints, "Constraints for a PoRep")
+    describe(porep_snark_partitions, "Number of partitions of a PoRep proof")
+    describe(porep_snark_size, "Size of a PoRep proof (bytes)")
+    describe(porep_snark_size_kb, "Size of a PoRep proof (KiB)")
+    describe(porep_snark_partition_constraints, "Number of constraints for a PoRep partition")
+    describe(kdf_constraints, "Constraints of KDF per node")
+
+    describe(labeling_proof, "Constraints to check labeling of a single node")
+    describe(labeling_proofs, "Constraints to check labeling of a window")
+    describe(window_labeling_proof, "Constraints to check correct labeling for all windows")
+    describe(data_inclusion, "Constraints to check a single inclusion in CommD")
+    describe(data_inclusions, "Constraints to check all the inclusions required in a window in CommD")
+    describe(window_data_inclusions,  "Constraints to check all the inclusions required for all windows in CommD")
+    describe(replica_inclusion, "Constraints to check a single inclusion in CommR")
+    describe(replica_inclusions, "Constraints to check all the inclusions required in a window in CommR")
+    describe(window_replica_inclusions,  "Constraints to check all the inclusions required for all windows in CommR")
+    describe(window_inclusion, "Constraints to check a single inclusion proof in a window")
+    describe(window_inclusions, "Constraints to check all the required inclusion proofs in a window")
+    describe(column_leaf, "Constraints to check a single leaf generation from a column in a window")
+    describe(column_leaves, "Constraints to check all the leaves generation required in a window")
+    describe(window_column_leaves, "Constraints to check all the leaves generation required for all windows")
+
+    // D
+    // Data layer committed with merkle tree
+    data_inclusion = merkle_tree_datahash_constraints * log2(nodes)
+    data_inclusions = data_inclusion * porep_challenges
+    window_data_inclusions = data_inclusion * porep_challenges * windows
```

Comment:
Should there be a factor of layers here? (Maybe this will be the same design question as before — just flagging as I read.)

Reply:
same as other comment